src/deepsparse/benchmark/benchmark_pipeline.py

```diff
@@ -0,0 +1,525 @@
+    scheduler = parse_scheduler(scenario)
+    num_streams = parse_num_streams(num_streams, num_cores, scenario)
+
+    input_type = config["data_type"]
+    kwargs = {}
+    if "pipeline_kwargs" in config:
+        kwargs = config["pipeline_kwargs"]
+    pipeline = Pipeline.create(
+        task=task,
+        model_path=model_path,
+        engine_type=engine,
+        scheduler=scheduler,
+        num_cores=num_cores,
+        **kwargs,
+    )
+    inputs = create_input_schema(pipeline, input_type, batch_size, config)
+
+    if scenario == "singlestream":
+        singlestream_benchmark(pipeline, inputs, warmup_time)
+        start_time = time.perf_counter()
+        batch_times = singlestream_benchmark(pipeline, inputs, seconds_to_run)
+    elif scenario == "multistream":
+        multistream_benchmark(pipeline, inputs, warmup_time, num_streams)
+        start_time = time.perf_counter()
+        batch_times = multistream_benchmark(
+            pipeline, inputs, seconds_to_run, num_streams
+        )
+    elif scenario == "elastic":
+        multistream_benchmark(pipeline, inputs, warmup_time, num_streams)
+        start_time = time.perf_counter()
+        batch_times = multistream_benchmark(
+            pipeline, inputs, seconds_to_run, num_streams
+        )
+    else:
+        raise Exception(f"Unknown scenario '{scenario}'")
+
+    end_time = time.perf_counter()
+    total_run_time = end_time - start_time
+    if len(batch_times) == 0:
+        raise Exception(
+            "Generated no batch timings, try extending benchmark time with '--time'"
+        )
+
+    return batch_times, total_run_time, num_streams
+
+
+def calculate_statistics(
+    batch_times_ms: List[float], total_run_time_ms: float, num_streams: int
+) -> Dict:
+    percentiles = [25.0, 50.0, 75.0, 90.0, 95.0, 99.0, 99.9]
+    buckets = numpy.percentile(batch_times_ms, percentiles).tolist()
+    percentiles_dict = {
+        "{:2.1f}%".format(key): value for key, value in zip(percentiles, buckets)
+    }
+
+    benchmark_dict = {
+        "total_percentage": sum(batch_times_ms) / total_run_time_ms * 100 * num_streams,
+        "median": numpy.median(batch_times_ms),
+        "mean": numpy.mean(batch_times_ms),
+        "std": numpy.std(batch_times_ms),
+        **percentiles_dict,
+    }
+    return benchmark_dict
+
+
+def calculate_section_stats(
+    batch_times: List[StagedTimer], total_run_time: float, num_streams: int
+) -> Dict[str, Dict]:
+    compute_sections = batch_times[0].stages
+    total_run_time_ms = total_run_time * 1000
+
+    sections = {}
+    for section in compute_sections:
+        section_times = [st.times[section] * 1000 for st in batch_times]
+        sections[section] = calculate_statistics(
+            section_times, total_run_time_ms, num_streams
+        )
+
+    return sections
+
+
+def main():
+    args = parse_args()
+    config = parse_input_config(args.input_config)
+
+    print("Original Model Path: {}".format(args.model_path))
```

Comment:
use the logger instead maybe?

Reply:
nit: use Old School style formatting for logger
https://realpython.com/python-f-strings/#how-to-use-formatting